benchmarks/vector_db_compare/mongodb_vector_bench.py

```
@@ -0,0 +1,414 @@
+    parser.add_argument("--top-k", type=int, default=10)
+    parser.add_argument("--search-concurrency", default="2,4,8,16,32,64,128")
+    parser.add_argument("--num-candidates", type=int, default=128)
+    parser.add_argument("--batch-size", type=int, default=1000)
+    parser.add_argument("--index-timeout-seconds", type=float, default=300)
+    parser.add_argument("--min-recall", type=float, default=0.95)
+    args = parser.parse_args()
+
+    dataset_dir = Path(args.dataset_dir)
+    manifest = load_manifest(dataset_dir)
+    if manifest["metric"] != "cosine" or not manifest["normalized"]:
+        raise RuntimeError(f"unsupported dataset metric/normalization: {manifest}")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)